File: python/basic_algorithm/former/twoSum.py
class Solution(object):

    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        newdict = dict()
        num = enumerate(nums)
        for x in num:
            newdict[x[1]] = x[0]
            if (target-x[1]) in newdict and x[0] != newdict.get(target-x[1]):
                return [newdict.get(target - x[1]),x[0]]


nums = [3,3]
target = 6
s = Solution()
# print(s.twoSum(nums,target))
dict2 = {3:2,5:4}

File: java/basic_algorithm/former/twoSum.java
import java.util.Map;
import java.util.HashMap;

public class twoSum{
     public static int[] twoSum(int[] nums, int target) {
         Map<Integer,Integer> map = new HashMap<Integer,Integer>();
         for(int i =0;i<nums.length;i++){
             map.put(nums[i],i);
             int remain = target - nums[i];
             if(map.containsKey(remain)&&map.get(remain)!=i){
                 return new int[]{map.get(remain),i};
             }
         }
        throw new IllegalArgumentException("No two sum solution");

}
   public static void main(String []args){
       int []array = {3,3};
       int target = 6;
       int []result = new int[2];
       result = twoSum(array,target);
       System.out.print(result[0]);
       System.out.print(result[1]);
   }
}
